### DISCUSSION TITLE
Firebase Data Migration to Supabase - 0 records written

### DISCUSSION THREAD

**Discussion by VincentLu91** [supabase/supabase#19474] [Questions]
# Bug report
Tried dumping Firestore collection but instead got 0 records and thus I could not import JSON file to Supabase (got an `]` character in the output json file instead.

## Describe the bug
This is supposed to be trivial but I couldn't migrate all the collections from Firebase to Supabase. I'm following the article below:
https://supabase.com/docs/guides/resources/migrating-to-supabase/firestore-data
I added `supabase-service.json` and `firebase-service.json` to the `firestore` folder.  However, I can't seem to dump the collections already. The `collections.js` script correctly I identified that I have 3 collections, but 0 records were written to the collections json file I created so I can't import the JSON to Supabase at all.

See bug report.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to https://supabase.com/docs/guides/resources/migrating-to-supabase/firestore-data
2. Follow the steps until you get to `Command line options` section
3. run `node collections.js`
4. run `node firestore2json.js my_collection`
5. At this point your output should look like:
```
(base) Vincents-MacBook-Pro:firestore vincentlu$ node collections.js
collection1
collection2
collection3
(base) Vincents-MacBook-Pro:firestore vincentlu$ node firestore2json.js my_collection
0 records written to my_collection.json
(base) Vincents-MacBook-Pro:firestore vincentlu$
```
6. If you open firestore/my_collection.json, the contents are near empty, just:
```

]
```

## Expected behavior

Json files generated should be properly populated!

## Screenshots

see the steps under `To Reproduce` section

## System information

- OS: macOS
- Browser (if applies) safari
- Version of supabase-js: N/A. I just used https://github.com/supabase-community/firebase-to-supabase
- Version of Node.js: v14.17.0

**Comment by VincentLu91** [comment]
Turns out I entered the wrong flag to the firestore2json.js commands. The flag should be a valid collection. Run:

```node collections.js```

Output (in my case):
```
customers
products
recordings
```

But one of the collections has subcollections, so when I ran that, I got nothing...
```
node firestore2json.js recordings
0 records written to recordings.json
```

Here's what my Firestore collections look like (see attached).

[
![Screenshot 2023-01-30 at 10 10 45 PM](https://user-images.githubusercontent.com/3411100/215655614-40d1878a-be55-4f87-98e1-cbb640cee7a4.png)
](url)

How do I convert the subcollections to json so that they are nested in the `recordings` collection??

**Reply by hstro-ai** [reply]
Did you find a solution to this? I am facing the same challenge.

**Reply by edinskeja** [reply]
@VincentLu91 did you manage to solve the subcollections issue?
